### PR TITLE
Changes for the travis-enterprise release

### PR DESCRIPTION
**Pull request by tyranja**
We have added application-level locks for aggregating logs.

### CODE REVIEW

**Comment by solarce** [comment]
Thanks for this, I don't think blue will able to review it until late February. Let me know if you need us to prioritize it sooner

File: lib/travis/logs/helpers/database.rb [hunk 1]
```diff
@@ -54,7 +54,7 @@ def self.jdbc_uri_from_config(config)
           params.merge!(
             ssl: true,
             sslfactory: 'org.postgresql.ssl.NonValidatingFactory'
-          ) unless %w(1 yes on).include?(ENV['PG_DISABLE_SSL'].to_s.downcase)
+          ) if config[:ssl]
```
Comment (joshk) [inline lib/travis/logs/helpers/database.rb]: shouldn't this be using https://github.com/travis-ci/travis-logs/pull/43/files#diff-13fa1867dbc7ecbff0f1d8be526dc13dR7 or https://github.com/travis-ci/travis-logs/pull/43/files#diff-13fa1867dbc7ecbff0f1d8be526dc13dR11 ?

**Comment by joshk** [comment]
All in all this looks good. I left one question which would be good to answer before merging this in.

**Comment by igorwwwwwwwwwwwwwwwwwwww** [comment]
is this PR still relevant?

**Comment by bnferguson** [comment]
Closing this. Started bringing in the changes in this PR more piecemeal:

- [X] SSL changes
- [ ] `travis-config` upgrade (may also include AMPQS changes) 
- [ ] Run locking around aggregating logs. 

Reasoning is at the run locking has some bugs in it currently in `enterprise-2.0` so bringing in the other unrelated changes lets me get `master` builds of Enterprise working while I also isolate what's breaking there.

**Pull request closed without merging**